spec/form_objects/support_ticket/check_your_request_form_spec.rb

```diff
@@ -19,16 +19,49 @@
       allow(Settings.zendesk).to receive(:token).and_return('123456')
     end
 
-    it 'calls the zendesk service to create a ticket in zendesk' do
-      allow(ZendeskService).to receive(:send!)
-      described_class.new(ticket: mock_ticket).create_ticket
-      expect(ZendeskService).to have_received(:send!)
+    describe '#create_ticket' do
+      it 'calls the zendesk service to create a ticket in zendesk' do
+        allow(ZendeskService).to receive(:send!).and_return(instance_double('ZendeskAPI::Request', id: 56_789))
+        described_class.new(ticket: mock_ticket).create_ticket
+        expect(ZendeskService).to have_received(:send!)
+      end
+
+      it 'sets ticket_number instance variable' do
+        allow(ZendeskService).to receive(:send!).and_return(instance_double('ZendeskAPI::Request', id: 56_789))
+        klass = described_class.new(ticket: mock_ticket)
+        klass.create_ticket
+        expect(klass.ticket_number).to eq(56_789)
+      end
+
+      it 'returns ticket_number created in zendesk' do
+        allow(ZendeskService).to receive(:send!).and_return(instance_double('ZendeskAPI::Request', id: 15_264))
+        klass = described_class.new(ticket: mock_ticket)
+        expect(klass.create_ticket).to eq(15_264)
+      end
+
+      describe 'when ZendeskService fails' do
+        # ZendeskAPI::Request.create! should return nil if there are any errors while trying to create zendesk ticket
+        it 'sets ticket_number instance variable to nil' do
+          allow(ZendeskService).to receive(:send!).and_return(nil)
+          klass = described_class.new(ticket: mock_ticket)
+          klass.create_ticket
+          expect(klass.ticket_number).to be_nil
+        end
+
+        it 'returns nil' do
+          allow(ZendeskService).to receive(:send!).and_return(nil)
+          klass = described_class.new(ticket: mock_ticket)
+          expect(klass.create_ticket).to be_nil
+        end
+      end
     end
 
     context 'Subject' do
       it 'sets the email subject line to include just school name' do
         allow(ZendeskService).to receive(:send!)
-        described_class.new(ticket: { 'school_name' => 'School 1' }).create_ticket
+
+        klass = described_class.new(ticket: { 'school_name' => 'School 1' })
```

Comment:
more of a style thing `klass` typically refers to an actual class and not an instance of a class. eg. I would expect `klass.new` to return me an instance